sdks/java/harness/src/main/java/org/apache/beam/fn/harness/control/ProcessBundleHandler.java

```diff
@@ -742,6 +749,7 @@ synchronized void setInstructionId(String instructionId) {
     }
 
     void reset() throws Exception {
+      this.instructionId = null;
```

Comment:
I think this broke Java PreCommit (it looks like this was submitted while precommits weren't running). Running `.gradlew :sdks:java:harness:spotbugsMain` on master right now yields:
```
Inconsistent synchronization of org.apache.beam.fn.harness.control.ProcessBundleHandler$BundleProcessor.instructionId; locked 66% of time                                                                                               
                                                                                
                                                                                
Bug type IS2_INCONSISTENT_SYNC (click for details)                              
In class org.apache.beam.fn.harness.control.ProcessBundleHandler$BundleProcessor
Field org.apache.beam.fn.harness.control.ProcessBundleHandler$BundleProcessor.instructionId
Synchronized 66% of the time                                                    
Unsynchronized access at ProcessBundleHandler.java:[line 752]                   
Synchronized access at ProcessBundleHandler.java:[line 748]                     
Synchronized access at ProcessBundleHandler.java:[line 744] 
```

Reply:
Opened https://github.com/apache/beam/pull/15085